### PR TITLE
Gallery + re-template + code style + overload

### PR DESCRIPTION
**Pull request by nabobalis**
Did you know it's citrus season in California?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -23,6 +23,9 @@ deps =
     oldestdeps: matplotlib==3.5.0
     figure-!devdeps-!oldestdeps: astropy==5.3.4
     figure-!devdeps-!oldestdeps: matplotlib==3.8.1
+    # Due to https://github.com/matplotlib/pytest-mpl/issues/216
+    # Needs a release of pytest-mpl with the fix
+    figure: pluggy<1.4
```
Comment (Cadair) [inline tox.ini:28]: but we don't have an issue with pytest-mpl?

Comment (nabobalis) [reply tox.ini:28]: True, I don't know why.

Comment (nabobalis) [reply tox.ini]: I removed it.

**Comment by nabobalis** [comment]
> Did you know it's citrus season in California?

There are so many lemons and oranges for free outside homes.

**Comment by nabobalis** [comment]
TIME FOR A LEMON CHEESECAKE

File: mpl_animators/line.py [hunk 2]
```diff
@@ -95,18 +110,11 @@ def __init__(self, data, plot_axis_index=-1, axis_ranges=None, ylabel=None, xlab
...
-            # If the shape of the array is a 1D array, get the centers about axis=0
-            if np.asarray(axis_ranges[self.plot_axis_index]).ndim == 1:
-                self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]), 0)
-
-            # Else derive the xdata as pixel centers from the pixel edges supplied by
-            # the user in axis_ranges[plot_axis_index] along axis=plot_axis_index
-            else:
-                self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]),
-                                                 plot_axis_index)
+            self.xdata = edges_to_centers_nd(np.asarray(axis_ranges[self.plot_axis_index]), plot_axis_index)
```
Comment (nabobalis) [inline mpl_animators/line.py:117]: This looks to be wrong.

**Comment by nabobalis** [comment]
Figure has changed due to different random data due to generator change.

**Comment by nabobalis** [comment]
SINCE THIS PR WAS NEVER GOING TO BE REVIWED OR MERGED.

**Pull request closed without merging**